File: Data Structures & Algorithms/validate-parentheses/submission-3.py
class Solution:
    def isValid(self, s: str) -> bool:
        mapper = { ")" : "(", "]" : "[", "}" : "{" }
        stack = []

        for c in s:
            if mapper.get(c):
                if stack and stack[-1] == mapper[c]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(c)
        return len(stack) == 0

File: Data Structures & Algorithms/daily-temperatures/submission-3.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        vector<int> res(n, 0);
        stack<int> st;
        int l = 0;
        for(int l = 0; l<n; l++) {
            while(!st.empty() && temperatures[st.top()] < temperatures[l]) {
                res[st.top()] = l - st.top();
                st.pop();
            }
            st.push(l);
        }
        return res;
    }
};
